Tempo/Views/LikedPostView.swift

```diff
@@ -0,0 +1,198 @@
+		
+		addConstraints([songTopConstraint, songWidthConstraint, songLeftConstraint, songRightConstraint, artistTopConstraint, artistLeftConstraint, artistWidthConstraint, artistRightConstraint])
+		
+	
+	}
+	
+	func updatePlayingStatus() {
+		updateSongLabel()
+		updateBackground()
+	}
+	
+	override func didMoveToWindow() {
+		
+		if tapGestureRecognizer == nil {
+			tapGestureRecognizer = UITapGestureRecognizer(target: self, action: #selector(likedPostViewPressed(_:)))
+			tapGestureRecognizer?.delegate = self
+			tapGestureRecognizer?.cancelsTouchesInView = false
+			addGestureRecognizer(tapGestureRecognizer!)
+		}
+		
+		if longPressGestureRecognizer == nil {
+			longPressGestureRecognizer = UILongPressGestureRecognizer(target: self, action: #selector(likedPostViewPressed(_:)))
+			longPressGestureRecognizer?.delegate = self
+			longPressGestureRecognizer?.minimumPressDuration = 0.5
+			longPressGestureRecognizer?.cancelsTouchesInView = false
+			addGestureRecognizer(longPressGestureRecognizer!)
+		}
+		
+		albumArtworkImageView?.clipsToBounds = true
+		isUserInteractionEnabled = true
+		
+		layer.borderColor = UIColor.tempoDarkGray.cgColor
+		layer.borderWidth = 0.7
+	}
+	
+	// Customize view to be able to re-use it for search results.
+	func flagAsSearchResultPost() {
+		songArtistLabel?.text = post!.song.title + " · " + post!.song.album
+	}
+	
+	func updateAddButton() {
+		addButton!.isHidden = true
+		if let _ = post {
+			SpotifyController.sharedController.spotifyIsAvailable { success in
+				if success {
+					self.addButton!.isHidden = false
+				}
+			}
+			if songStatus == .saved {
+				self.addButton?.setBackgroundImage(UIImage(named: "check"), for: .normal)
+			} else {
+				self.addButton?.setBackgroundImage(UIImage(named: "plus"), for: .normal)
+			}
+		}
+	}
+	
+	func updateSongLabel() {
+		if let post = post {
+			let color: UIColor
+			let duration = TimeInterval(0.3)
+			if post.player.isPlaying {
+				color = UIColor.tempoLightRed
+			} else {
+				color = UIColor.white
+			}
+			
+			guard let label = songNameLabel else { return }
+			if !label.textColor.isEqual(color) {
+				UIView.transition(with: label, duration: duration, options: .transitionCrossDissolve, animations: {
+					label.textColor = color
+				}, completion: { _ in
+					label.textColor = color
+				})
+			}
+		}
+	}
+	
+	func updateBackground() {
+		if let post = post {
+			backgroundColor = post.player.isPlaying ? UIColor.tempoDarkGray : UIColor.tempoLightGray
+		}
+	}
+	
+	func likedPostViewPressed(_ sender: UIGestureRecognizer) {
+		guard let _ = post else { return }
+		
+		if sender is UITapGestureRecognizer {
+			let tapPoint = sender.location(in: self)
+			let hitView = hitTest(tapPoint, with: nil)
+			if hitView == addButton {
+				if songStatus == .notSaved {
+					SpotifyController.sharedController.saveSpotifyTrack(post!) { success in
+						if success {
+							self.songStatus = .saved
```

Comment:
waaaay too indented, flatten out the indentation by using multiple boolean conditions in one if statement